mlir/include/Quantum/IR/QuantumInterfaces.td

```diff
@@ -64,17 +64,25 @@ def QuantumGate : OpInterface<"QuantumGate"> {
 
     let methods = [
         InterfaceMethod<
-            "Return all operands which are considered input qubit values.",
+            "Return all operands which are considered input qubit values (including controls).",
             "std::vector<mlir::Value>", "getQubitOperands"
         >,
         InterfaceMethod<
-            "Return all results which are considered output qubit values.",
+            "Set all operands which are considered input qubit values (including controls).",
+            "void", "setQubitOperands", (ins "mlir::ValueRange":$replacements)
+        >,
+        InterfaceMethod<
+            "Return all results which are considered output qubit values (including controls).",
             "std::vector<mlir::Value>", "getQubitResults"
```

Comment:
Would it also make sense to have a set for the results? I think it can be potentially

Reply:
Results are immutable in MLIR, you have to create a new op to create new values